contracts/SmartLoan.sol

```diff
@@ -48,6 +53,105 @@ contract SmartLoan is OwnableUpgradeable, PriceAwareUpgradeable {
+    MIN_SELLOUT_LTV = _newMinSelloutLtv;
+  }
+
+
+  /**
+   * This function allows selling assets without checking if the loan will remain solvent after this operation.
+   * It is used as part of the sellout() function which sells part/all of assets in order to bring the loan back to solvency.
+   * It is possible that multiple different assets will have to be sold and for that reason we do not use the remainsSolvent modifier.
+  **/
+  function nonSolventAssetSale(bytes32 asset, uint256 _amount, uint256 _minAvaxOut) private {
+    IERC20Metadata token = getERC20TokenInstance(asset);
+    token.transfer(address(exchange), _amount);
+
+    (bool success,) = address(exchange).call{value : 0}(
+      abi.encodeWithSignature("sellAsset(bytes32,uint256,uint256)", asset, _amount, _minAvaxOut)
+    );
+    if (!success) {
+      exchange.transferBack(asset);
+    }
+  }
+
+
+  /**
+  * This function attempts to sell just enough asset to receive targetAvaxAmount.
+  * If there is not enough asset's balance to cover the whole targetAvaxAmount then the whole asset's balance
+  * is being sold.
+  * It is possible that multiple different assets will have to be sold and for that reason we do not use the remainsSolvent modifier.
+  **/
+  function nonSolventPartialOrFullAssetSale(bytes32 asset, uint256 targetAvaxAmount) private {
+    IERC20Metadata token = getERC20TokenInstance(asset);
+    uint256 balance = token.balanceOf(address(this));
+    if (balance > 0) {
+      uint256 saleAvaxValue = exchange.getEstimatedAVAXFromERC20Token(balance, supportedAssets.getAssetAddress(asset));
+      if (saleAvaxValue < targetAvaxAmount) {
+        nonSolventAssetSale(asset, balance, saleAvaxValue);
+      } else {
+        uint256 saleAmount = balance * targetAvaxAmount / saleAvaxValue;
+        nonSolventAssetSale(asset, saleAmount, targetAvaxAmount);
+      }
+    }
+  }
+
+
+  /**
+  * This function attempts to repay the _repayAmount back to the pool as well as pay a bonus to liquidator.
+  * If there is not enough AVAX balance to repay the _repayAmount then the available AVAX balance will be repaid and no
+  * liquidation bonus will be paid to the liquidator.
+  **/
+  function attemptRepay(uint256 _repayAmount) internal {
+    if (address(this).balance < _repayAmount) {
+      repay(address(this).balance);
+    } else {
+      repay(_repayAmount);
+      uint256 bonus = _repayAmount * LIQUIDATION_BONUS / PERCENTAGE_PRECISION;
+      if (bonus < address(this).balance) {
+        payable(msg.sender).transfer(bonus);
+      } else {
+        payable(msg.sender).transfer(address(this).balance);
+      }
+
+    }
+  }
+
+
+  /**
+  * This function role is to sell part/all of the available assets in order to bring the loan back to a solvent state.
+  *
+  **/
+  function sellout(uint256 repayAmount) external successfullSellout {
+    require(!isSolvent(), "Cannot sellout a solvent account");
+    uint256 debt = getDebt();
+    if (repayAmount > debt) {
```

Comment:
Yes, it's necessary so that the sellout can succeed instead of being reverted with an error stating that we are trying to repay more that was actually borrowed